presto-main/src/main/java/com/facebook/presto/memory/MemoryPool.java

```diff
@@ -49,25 +49,19 @@
     private final MemoryPoolId id;
     private final long maxBytes;
 
-    @GuardedBy("this")
-    private long reservedBytes;
-    @GuardedBy("this")
-    private long reservedRevocableBytes;
+    private volatile long reservedBytes;
```

Comment:
Does it make sense to mark it with `@GuardedBy("this")`? cc @aweisberg , @arhimondr